### PR TITLE
Move built-in, external string, and interpreter globals to the global context

### PR DESCRIPTION
**Pull request by zherczeg**
There are still globals left.

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 1]
```diff
@@ -798,56 +838,6 @@ ecma_builtin_dispatch_construct (ecma_object_t *obj_p, /**< built-in object */
 } /* ecma_builtin_dispatch_construct */
 
 /**
- * Dispatcher of built-in routines
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_dispatch_routine (ecma_builtin_id_t builtin_object_id, /**< built-in object' identifier */
```
Comment (galpeter) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c]: If I see this correctly this was just moved above a few lines. Do we really need this to be moved?

Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c]: Not necessary, but I would prefer to do this. It is better from logical perspective.

**Comment by dbatyai** [comment]
LGTM

File: jerry-core/parser/regexp/re-compiler.c [hunk 2]
```diff
@@ -614,25 +612,26 @@ re_compile_bytecode (const re_compiled_code_t **out_bytecode_p, /**< [out] point
...
+
+      cache_idx = JERRY_CONTEXT (re_cache_idx)++;
 
-      if (cached_bytecode_p != NULL)
+      /* The garbage collector might run during the byte code
+       * allocations above and it may free this entry. */
+      if (JERRY_CONTEXT (re_cache)[cache_idx] != NULL)
       {
-        ecma_bytecode_deref ((ecma_compiled_code_t *) cached_bytecode_p);
+        ecma_bytecode_deref ((ecma_compiled_code_t *) JERRY_CONTEXT (re_cache)[cache_idx]);
```
Comment (LaszloLango) [inline jerry-core/parser/regexp/re-compiler.c]: style of `JERRY_CONTEXT (...)[...]` is mixed. There is a space after `)` in some place. I prefer this style (without space). Please unify that.

**Comment by LaszloLango** [comment]
LGTM after the style is fixed.

**Pull request merged**